contracts/oracles/fBEETSPriceOracle.sol

```diff
@@ -2,48 +2,70 @@
 
 import "hardhat/console.sol";
 
+interface IFBeets {
+    function vestingToken() external view returns (address);
+}
+
 contract fBEETSPriceOracle is SpotPriceOracle {
     using SafeMath for uint256;
+    using Address for address;
 
     address vestingToken;
     address ftmUsdOracle;
     address fBeetsBar;
 
-    constructor(
-        address _vestingToken,
-        address _ftmUsdOracle,
-        address _fBeetsBar
-    ) SpotPriceOracle() {
-        vestingToken = _vestingToken;
+    constructor(address _fBeetsBar) SpotPriceOracle() {
+        require(_fBeetsBar.isContract(), "fBeetsBar is not a contract");
         ftmUsdOracle = _ftmUsdOracle;
         fBeetsBar = _fBeetsBar;
+        vestingToken = IFBeets(_fBeetsBar).vestingToken();
     }
 
-    function getPrice() public view override returns (int256) {
-        IBalV2PriceOracle.OracleAverageQuery[]
-            memory query = new IBalV2PriceOracle.OracleAverageQuery[](1);
-        query[0] = IBalV2PriceOracle.OracleAverageQuery(
-            IBalV2PriceOracle.TWAP_VALUE.BPT_PRICE,
-            120,
-            10
-        );
+    function updateSafePrice() public payable returns (int256) {
+       return getSafePriceInternal();
+    }
+
+    function getSafePrice() public view override returns (int256) {
+       return getSafePriceInternal();
+    }
+
+    function getSafePriceInternal() internal view returns (int256) {
+      IBalV2PriceOracle.OracleAverageQuery[]
+          memory query = new IBalV2PriceOracle.OracleAverageQuery[](1);
+      query[0] = IBalV2PriceOracle.OracleAverageQuery(
+          IBalV2PriceOracle.TWAP_VALUE.BPT_PRICE,
+          120,
+          10
+      );
+      uint256[] memory prices = IBalV2PriceOracle(vestingToken).getTimeWeightedAverage(
+          query
+      );
+      uint256 bptFtm = prices[0];
+      uint256 fBeetsTotalSupply = IERC20(fBeetsBar).totalSupply();
+      uint256 vestingTokenLocked = IERC20(vestingToken).balanceOf(fBeetsBar);
+      uint256 bptFBeetsRatio = vestingTokenLocked.mul(1e18).div(fBeetsTotalSupply);
+      uint256 price = bptFtm.mul(bptFBeetsRatio);//.div(1e36);
+      return int256(price);
+    }
+
+    // fast but vulnerable to flashloan attacks
+    function getCurrentPrice() public view override returns (int256) {
         uint256[] memory prices = IBalV2PriceOracle(vestingToken).getTimeWeightedAverage(
-            query
+            IBalV2PriceOracle.TWAP_VALUE.BPT_PRICE
         );
         uint256 bptFtm = prices[0];
-        int256 ftmUsd = AggregatorV3Interface(ftmUsdOracle).latestAnswer();
         uint256 fBeetsTotalSupply = IERC20(fBeetsBar).totalSupply();
         uint256 vestingTokenLocked = IERC20(vestingToken).balanceOf(fBeetsBar);
         uint256 bptFBeetsRatio = vestingTokenLocked.mul(1e18).div(fBeetsTotalSupply);
-        uint256 price = bptFtm.mul(bptFBeetsRatio).mul(uint256(ftmUsd)).div(1e36);
+        uint256 price = bptFtm.mul(bptFBeetsRatio);
         return int256(price);
     }
```

Comment:
this can return the same results than getSafePrice. getCurrentPrice _can_ be vulnerable to flashloans, but doesn't mean it has to. It will reduce code duplication